### DISCUSSION TITLE
How to create a logs

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#2427] [Help]
Im using MagickWand methods to resize the Image 
How to handle to exception
How to create a log ...

**Comment by programmerarunkumar** [comment]
> Im using MagickWand methods to resize the Image
How to handle to exception
How to create a log 



void resize(char *oldImage,char *newImage){ 

  MagickWand *magick_wand;
  MagickBooleanType status;

  magick_wand=NewMagickWand();//Initialize the MagickWand
  printf("Method : NewMagickWand() completed,%s\n",oldImage);

  MagickReadImage(magick_wand,oldImage);//Read the Image
  printf("Method : MagickReadImage() completed,%s\n",oldImage);

  MagickResizeImage(magick_wand,1600,1600,LanczosFilter);//Resize the image
  printf("Method : MagickResizeImage() completed,%s\n",oldImage);

  MagickWriteImage(magick_wand,newImage);//write the image
  printf("Method : MagickWriteImage() completed,%s\n",oldImage);

  DestroyMagickWand(magick_wand);//Destroy the MagickWand
  printf("Method : DestroyMagickWand() completed,%s\n",oldImage);


}	

This is My code to resize the image , How to add logs for it

**Reply by programmerarunkumar** [reply]
> > Im using MagickWand methods to resize the Image
> > How to handle to exception
> > How to create a log
> 
> void resize(char *oldImage,char *newImage){
> 
> MagickWand *magick_wand;
> MagickBooleanType status;
> 
> magick_wand=NewMagickWand();//Initialize the MagickWand
> printf("Method : NewMagickWand() completed,%s\n",oldImage);
> 
> MagickReadImage(magick_wand,oldImage);//Read the Image
> printf("Method : MagickReadImage() completed,%s\n",oldImage);
> 
> MagickResizeImage(magick_wand,1600,1600,LanczosFilter);//Resize the image
> printf("Method : MagickResizeImage() completed,%s\n",oldImage);
> 
> MagickWriteImage(magick_wand,newImage);//write the image
> printf("Method : MagickWriteImage() completed,%s\n",oldImage);
> 
> DestroyMagickWand(magick_wand);//Destroy the MagickWand
> printf("Method : DestroyMagickWand() completed,%s\n",oldImage);
> 
> }
> 
> This is My code to resize the image , How to add logs for it

I have added print after each method
I need to create a log file for each resize call with that prints in my C code
how to do it ?

**Comment by snibgo** [comment]
Call SetLogEventMask() with a parameter such as "all".

**Reply by programmerarunkumar** [reply]
> Call SetLogEventMask() with a parameter such as "all".

Where I can know the all method related to log.
any resources ...

**Reply by programmerarunkumar** [reply]
> Call SetLogEventMask() with a parameter such as "all".

How to know more about this SetLogEventMask() method such as parameter ,return type

**Reply by snibgo** [reply]
See the source code in log.c

**Reply by programmerarunkumar** [reply]
> See the source code in log.c

Im using MagickWand
log.c is MagickCore 
will able to use  SetLogEventMask() in my C code(using MagickWand method in C ) ?

**Comment by snibgo** [comment]
You can list the methods with "magick -list debug". See also https://www.imagemagick.org/script/command-line-options.php#debug
